src-web/components/kappnav/common/ResourceTable.js

```diff
@@ -117,6 +117,14 @@ class ResourceTable extends React.Component {
 	   those resources which doesn't have the "section-map" attribute or if the "section-map" 
 	   attribute is empty*/
 	componentDidMount() {
+		//If the resource table is empty, a row with text should have a link to the 'Add [resoure_name]' navmodal
```

Comment:
Since the modal click behavior needs to be done in separate places, create one method for this behavior (something like connectAddLinkOnclick) and call it from both componentDidMount and componentDidUpdate.

Reply:
Refactored the code and created the suggested connectAddLinkOnclick() method.